src/library/scala/collection/immutable/ChampCommon.scala

```diff
@@ -49,19 +51,19 @@ private[immutable] abstract class Node[T <: Node[T]] {
 
   def sizePredicate: Int
 
-  protected final def removeElement(as: Array[Int], ix: Int): Array[Int] = {
+  protected final def removeElement[A: ClassTag](as: Array[A], ix: Int): Array[A] = {
     if (ix < 0) throw new ArrayIndexOutOfBoundsException
     if (ix > as.length - 1) throw new ArrayIndexOutOfBoundsException
-    val result = new Array[Int](as.length - 1)
+    val result = new Array[A](as.length - 1)
     arraycopy(as, 0, result, 0, ix)
     arraycopy(as, ix + 1, result, ix, as.length - ix - 1)
     result
   }
 
-  protected final def insertElement(as: Array[Int], ix: Int, elem: Int): Array[Int] = {
+  protected final def insertElement[A: ClassTag](as: Array[A], ix: Int, elem: A): Array[A] = {
     if (ix < 0) throw new ArrayIndexOutOfBoundsException
     if (ix > as.length) throw new ArrayIndexOutOfBoundsException
-    val result = new Array[Int](as.length + 1)
+    val result = new Array[A](as.length + 1)
```

Comment:
I'm hesitant to use generic array operations here without benchmarking against the alternatives. The possible costs I'm worried about are creation of the `ClassTag` to wrap the `Class` constant, and making the array operations go through the indirection of `Array_update` etc.

Reply:
This was done so that the method could be used for content as well as indexes/hashes. I've now reverted this to `Array[Int]` specific, and made a new `removeAnyElement` for the `Array[Any]` case